### PR TITLE
fix: support multiple versions in pull command

### PR DESCRIPTION
**Pull request by bloodearnest**
- Add opensafely label in image filters.
- Support v2 python in opensafely pull

### CODE REVIEW

File: opensafely/pull.py [hunk 1]
```diff
@@ -104,19 +125,22 @@ def get_local_images():
...
     for line in ps.stdout.splitlines():
         if not line.strip():
             continue
 
-        name, sha = line.split("=", 1)
-        images[name].append(sha)
+        line = line.replace("ghcr.io/opensafely-core/", "")
+
+        image, sha = line.split("=", 1)
+        assert image not in images
```
Comment (ghickman) [inline opensafely/pull.py:142]: Is this likely to happen?  If so can we throw something more meaningful for the user?  I'm thinking a note to contact tech support is likely more than enough!

File: opensafely/pull.py [hunk 2]
```diff
@@ -133,6 +157,8 @@ def remove_deprecated_images(local_images):
 def get_default_version_for_image(name):
     if name in ["databuilder", "ehrql"]:
         return "v0"
+    elif name == "python":
+        return "v2"
     else:
         return "latest"
```
Comment (ghickman) [inline opensafely/pull.py:163]: You could absolutely use a case statement here!  It would improve just about nothing but you _could_ 😜

**Review by ghickman — APPROVED** [review]

File: opensafely/pull.py (hunk 2 above)
Comment (bloodearnest) [reply opensafely/pull.py:163]: Sadly not, as opensafely-cli runs on user's installed python, so we have to support old python (current lower bound is 3.9, IIRC)

File: opensafely/pull.py (hunk 1 above)
Comment (bloodearnest) [reply opensafely/pull.py:142]: It's not at all likely to happen. I threw it in to check it wasn't, I'll remove it, thanks

File: opensafely/pull.py (hunk 2 above)
Comment (ghickman) [reply opensafely/pull.py:163]: booooo (but totally understandable)

**Pull request merged**